src/app/models/project.ts

```diff
@@ -73,6 +74,7 @@ export class Project {
     this.region = obj && obj.region || null;
     this.projectDirector = obj && obj.projectDirector || null;
     this.agreements = obj && obj.agreements || null;
+    this.shapefiles = obj && obj.shapefiles || [];
```

Comment:
what if obj.shapefiles is not an array, but something else will your code handle this, or will it always be an array, if it makes it to this point?